legal-api/report-templates/template-parts/special-resolution/resolutionApplication.html

```diff
@@ -0,0 +1,77 @@
+{% if toLegalName %}
+<div class="no-page-break">
+  <div class="separator mt-4"></div>
+  <div class="section-title mt-4">Change of Business Name</div>
+</div>
+<table class="section-data business-change-table mt-4" role="presentation">
+    <tr class="no-page-break">
+        <td class="col-50">
+            <div class="section-sub-title">From</div>
+            <div class="{{'upper-text' if fromLegalName!='Not Available'}} pt-2">{{ fromLegalName }}</div>
+        </td>
+        <td class="col-50">
+            <div class="section-sub-title">To</div>
+            <div class="{{'upper-text' if toLegalName!='Not Available'}} pt-2">{{ toLegalName }}</div>
+        </td>
+    </tr>
+    <tr>
+        <td class="section-data" colspan="2">
+            <div class="section-sub-title">Name Request Number</div>
+            <div class="{{'upper-text' if nrNumber!='Not Available'}} pt-2">{{ nrNumber }}</div>
+        </td>
+    </tr>
+</table>
+<div class="separator mt-4"></div>
+{% endif %}
+
+{% if prevCoopAssociationType %}
+<div class="no-page-break">
+  <div class="separator mt-4"></div>
+  <div class="section-title mt-4">Change of Association Type</div>
+</div>
+<table class="section-data business-change-table mt-4" role="presentation">
+    <tr class="no-page-break">
+        <td class="col-50">
+            <div class="section-sub-title">From</div>
+            <div class="{{'upper-text' if prevCoopAssociationType!='Not Available'}} pt-2">{{ prevCoopAssociationType }}</div>
+        </td>
+        <td class="col-50">
+            <div class="section-sub-title">To</div>
+            <div class="{{'upper-text' if newCoopAssociationType!='Not Available'}} pt-2">{{ newCoopAssociationType }}</div>
+        </td>
+    </tr>
+</table>
+<div class="separator mt-4"></div>
+{% endif %}
+
+{% if rulesFileKey %}
```

Comment:
I don't think this works.  `rulesFileKey` is not always available.  Her is an example of a complete SR filing I found in DEV.

``` json
{
  "filing": {
    "alteration": {
      "business": {
        "identifier": "CP1002592",
        "legalType": "CP"
      },
      "cooperativeAssociationType": "CP",
      "memorandumFileKey": "8840b68c-aa6d-4bff-b5f8-8da3ef41ed49.pdf",
      "memorandumFileName": "CP1002592 - Certified Memorandum - 2023-06-23.pdf",
      "rulesInResolution": true
    },
    "business": {...},
    "header": {
      "name": "specialResolution",
      ...
    },
    "specialResolution": {
      "resolution": "<p>update rulesssss</p>",
      "resolutionConfirmed": true,
      "resolutionDate": "2023-06-23",
      "signatory": {
        "additionalName": "",
        "familyName": "snow",
        "givenName": "john"
      },
      "signingDate": "2023-06-23"
    }
  }
}

```

Reply:
I confirmed and fixed it

Reply:
i don't see `uploadNewRules` being used in `alteration` in any of the filings in DEV.  Are you sure you aren't trying to use `rulesFileKey`?

This is what I see for a SR filing in DEV when a rules file is uploaded.

``` json
{
  "filing": {
    "header": {
      "date": "2023-06-23",
      "name": "specialResolution",
      ...
    },
    "business": {
      "legalName": "SUPER SUPER COOP",
      "legalType": "CP",
      "identifier": "CP1002552",
      "foundingDate": "2022-07-27T18:45:39.188112+00:00"
    },
    "alteration": {
      "business": { ... },
      "contactPoint": { ... },
      "rulesFileKey": "304bf897-06ef-4868-94d7-c50419cc366f.pdf",
      "rulesFileName": "CP1002552 - Certified Rules - 2023-06-22.pdf",
      "rulesUploadedOn": "2023-06-22T14:43:18.579341+00:00",
      "memorandumFileKey": "test",
      "memorandumFileName": "CP1002552 - Certified Memorandum - 2023-05-26.pdf",
      "cooperativeAssociationType": "CP"
    },
    "specialResolution": {
      "signatory": {
        "givenName": "gsg",
        "familyName": "sgs",
        "additionalName": ""
      },
      "resolution": "<p>gsgsgs</p>",
      "signingDate": "2023-06-23",
      "resolutionDate": "2023-06-16",
      "resolutionConfirmed": true
    }
  }
}

```

Reply:
Hi @argush3 
https://github.com/bcgov/lear/blob/3db2b5a84cbf227fbb274723f6b74046f8ba6943/queue_services/entity-filer/src/entity_filer/filing_processors/alteration.py#L98
`uploadNewRules` is in `meta_data`, not in the `alteration` filing

Reply:
Ok, I see.  

But `{% if memorandumInResolution or uploadNewRules %}` should be `
{% if rulesInResolution or uploadNewRules %}` right?

Reply:
Yes. right. I was wrong.
Just fixed